transform/mattermost-analytics/models/marts/release/fct_issues_daily_snapshot.sql

```diff
@@ -1,14 +1,44 @@
+with release_versions as (
+    -- Keep releases
+    select
+        version,
+        short_version,
+        rc1_date,
+        planned_release_date,
+        actual_release_date,
+        -- Calculate 17th of month before the release's month. This is required as release timeframe is 17th to planned
+        -- release date.
+        dateadd(month, -1, planned_release_date) as _month_before_release,
```

Comment:
this prob. need documenting. For `_month_before_release` we use `planned_release_date` and for `week_after_release` we are using the `actual_release_date`

Reply:
It is already documented in https://mattermost.atlassian.net/wiki/spaces/DATAENG/pages/2490171393/Release+Quality. It is also documented in the additions in `_release_models.yml`. Do you think something extra is needed?

Reply:
Documentation was an extra reason for moving this logic from looker to DBT. Things seem easier to be documented here.

Reply:
since we do the `coalesce(actual_release_date::date, planned_release_date) as actual_release_date` in the staging layer, maybe it makes sense to use `actual_release_date` exclusively?

Reply:
No, it's two different things. 

- Planned release date is the date that the release is originally planned for. It is always present in the CSV.
- Actual release date is the date the release happened. It might >=  than the planned release date.  If it's `=`, then it's considered a normal release. If it's `>`, then it's considered a late release.
- CSV file contains some of the entries for actual dates. Some of them are equal to the planned release date, some are not and some are empty. For the ones that are empty, we consider the actual release date as equal to planned release date. 
- There's three important timeframes.
  - 17th to planned release date.
  - RC1 cut date to planned release date (the last part of the previous one).
  - A week after actual release date. Note that depending on whether a release is late or not, there might be a "gap" between planned release date and this timeframe.

The diagram in the [confluence](https://mattermost.atlassian.net/wiki/spaces/DATAENG/pages/2490171393/Release+Quality#Language) might also help understand this.

Reply:
ok, I think I get it - it is by design. I think documentation is great, should've read it before.